src/mscorlib/src/System/Char.cs

```diff
@@ -163,9 +163,13 @@ public bool Equals(Char obj)
       ==============================================================================*/
       // Provides a string representation of a character.
       [Pure]
-      public static String ToString(char c) {
+      public static String ToString(char c)
+      {
           Contract.Ensures(Contract.Result<String>() != null);
-          return new String(c, 1);
+
+          string result = string.FastAllocateString(1);
+          result.m_firstChar = c;
```

Comment:
@benaadams using `fixed (ptr = str)` generates less efficient code than `fixed (ptr = &str.m_firstChar)`. Also fixed itself also adds a bit of overhead with pinning the object.

I think what I'll do is simply add a method on `String` that does the work, e.g. `String.CreateFromChar(char)`, and defer to that.